### ISSUE TITLE
Delete with join on postgres produces invalid sql when providing multiple join conditions

### ISSUE THREAD

**Issue by JQuezada0** [knex/knex#5015]
# Environment

Knex version: 1.0.1
Database + version: Postgres 12
OS: MacOS 12.0.1 (Monterey)

# Bug

The docs give an example of how you can do a delete with a join, which produces valid SQL on both MySql and PostgreSql. But under certain conditions, it produces invalid sql when the dialect is Postgres.

All examples in this issue are also implemented here https://replit.com/join/izyqucnvdg-jquezada0

### Knex
```TS
knex('accounts')
 .where('activated', false)
 .join('accounts', 'accounts.id', 'users.account_id')
 .del()
 .toString()
```

### Output - Mysql ✅ 
```SQL
DELETE `accounts` FROM `accounts`
	INNER JOIN `accounts` ON `accounts`.`id` = `users`.`account_id`
	WHERE `activated` = FALSE;
```

### Output - Postgres ✅ 
```SQL
DELETE FROM "accounts" USING "accounts"
WHERE "activated" = FALSE
	AND "accounts"."id" = "users"."account_id";
```

### Knex
```TS
knex('accounts')
 .join('accounts', 'accounts.id', 'users.account_id')
 .del()
 .toString()
``` 

### Output - Mysql ✅ 
```SQL
DELETE `accounts` FROM `accounts`
	INNER JOIN `accounts` ON `accounts`.`id` = `users`.`account_id`;
```

### Output - Postgres ❌ 
```SQL
DELETE FROM "accounts" USING "accounts" "accounts"."id" = "users"."account_id";
```

- ***Notice the missing `WHERE`***

### Knex
```TS
/**
 * Keep the Where, and provide multiple join tables.
 */
 knex('accounts')
 .where('activated', false)
 .join('accounts', 'accounts.id', 'users.account_id')
 .join('settings', 'settings.id', 'users.settings_id')
 .del()
```

### Output - Mysql ✅ 
```SQL
DELETE `accounts` FROM `accounts`
	INNER JOIN `accounts` ON `accounts`.`id` = `users`.`account_id`
	INNER JOIN `settings` ON `settings`.`id` = `users`.`settings_id`
	WHERE `activated` = FALSE;
```

### Output - Postgres ✅ 
```SQL
DELETE FROM "accounts" USING "accounts", "settings"
WHERE "activated" = FALSE
	AND "accounts"."id" = "users"."account_id"
	AND "settings"."id" = "users"."settings_id";
```

### Knex
```TS
/**
 * Whether the Where is there or not, as soon
 * as you provide multiple join conditions on a single
 * table, provides invalid sql
 */
knex('accounts')
 .where('activated', false)
 .join('accounts', function(jb) {
   jb.on("accounts.id", "users.account_id")
   .andOn("accounts.user_id", "users.id")
 })
 .del()
```

### Output - Mysql ✅ 
```SQL
DELETE `accounts` FROM `accounts`
	INNER JOIN `accounts` ON `accounts`.`id` = `users`.`account_id`
		AND `accounts`.`user_id` = `users`.`id`
	WHERE `activated` = FALSE;
```

### Output - Postgres ❌ 
```SQL
DELETE FROM "accounts" USING "accounts"
WHERE "activated" = FALSE
	AND "accounts"."id" = "users"."account_id" "accounts"."user_id" = "users"."id";
```

- ***Notice the missing `AND` between the join conditions***
- ***If you omit the `WHERE` clause, it's still broken, the issue is the missing `AND` between the join conditions***

# Missing / erroneus documentation

Although the exact example given in the documentation works, the examples I've pasted above do not work, and maybe should be shown in the documentation (join with delete without where clause specifically)

# Expected behavior

### Knex
```TS
knex('accounts')
 .where('activated', false)
 .join('accounts', function(jb) {
   jb.on("accounts.id", "users.account_id")
   .andOn("accounts.user_id", "users.id")
 })
 .del()
```

### Output - Postgres ✅ 
```SQL
DELETE FROM "accounts" USING "accounts"
WHERE "activated" = FALSE
	AND "accounts"."id" = "users"."account_id" AND "accounts"."user_id" = "users"."id";
```

# Workaround

Here's a workaround until this can be fixed. But note, this workaround actually produces different sql on Mysql than would have been produced had I not tried to work around the form that causes invalid sql on postgres.

### Knex
```TS
knex('accounts')
 .where('activated', false)
 .join('accounts', function(jb) {})
 .andWhere({
   "accounts.id": knex.ref("users.account_id"),
   "accounts.user_id": knex.ref("users.id")
 })
 .del()
```

### Output - Mysql ✅ 
```SQL
DELETE `accounts` FROM `accounts`
	INNER JOIN `accounts`
	WHERE `activated` = FALSE
		AND `accounts`.`id` = `users`.`account_id`
		AND `accounts`.`user_id` = `users`.`id`;
```

### Output - Postgres ✅ 
```SQL
DELETE FROM "accounts" USING "accounts"
WHERE "activated" = FALSE
	AND "accounts"."id" = "users"."account_id"
	AND "accounts"."user_id" = "users"."id";
```

**Comment by kibertoad** [comment]
@OlivierCavadenti

**Comment by kibertoad** [comment]
Released in 1.0.3
